app/routes/goal_route.py

```diff
@@ -0,0 +1,132 @@
+from flask import Blueprint, make_response, request, jsonify
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+
+
+# Blueprint
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+# Helper functions
+def get_goal_with_goal_id(goal_id):
+    return Goal.query.get_or_404(goal_id, description={"details": "Invalid data"})
```

Comment:
By default `get_or_404` returns HTML not JSON.  I think you probably want to do something like this:

```suggestion
def get_goal_with_goal_id(goal_id):
    goal = Goal.query.get_or_404(goal_id)
    if goal:
        return goal
    else:
        abort(make_response({"details": "Invalid data"}, 404))
```